SwitchSpeak-iOS/SwitchSpeak-iOS/TouchGrid/TouchGrid.swift

```diff
@@ -140,15 +143,45 @@ class TouchGrid {
 		case .ROW_COLUMN:
 			for i in 0...(rows - 1) {
 				for j in 0...(cols - 1) {
-					(self.getRootNode()?.childNodes[i].childNodes[j] as? ButtonNode)?.button.setTitle(phrases[i * cols + j], for: .normal)
+					let buttonNode = self.getRootNode()?.childNodes[i].childNodes[j]
+					(buttonNode as? ButtonNode)?.button.setTitle(phrases[i * cols + j], for: .normal)
+					self.setDummy(node: buttonNode!)
 				}
+				self.setDummy(node: (self.getRootNode()?.childNodes[i])!)
 			}
 			break
 		case .BINARY_TREE:
 			break
 		case .LINEAR:
 			for i in 0...(rows * cols - 1) {
-				(self.getRootNode()?.childNodes[i] as? ButtonNode)?.button.setTitle(phrases[i], for: .normal)
+				let buttonNode = self.getRootNode()?.childNodes[i]
+				(buttonNode as? ButtonNode)?.button.setTitle(phrases[i], for: .normal)
+				self.setDummy(node: buttonNode!)
+			}
+		}
+	}
+	
+	/*
+	* 	sets the dummy variable for input node
+	*	if input node is a leaf node (i.e. a button node), check if title is "---"
+	*	if input node is a non-leaf node check if all its chidren are dummy nodes
+	*/
+	func setDummy(node: Node) {
```

Comment:
Can we change this to something like `determineDummy`? The first time I read `setDummy` I thought it was just setting the flag to true.